src/Conways/Game/Of/Life/GridFrame.java

```diff
@@ -4,20 +4,34 @@
 import java.awt.*;
 
 public class GridFrame extends JFrame {
+    Grid grid;
+    JButton nextButton;
+    JButton clearButton;
+    JPanel bottom = new JPanel();
 
+    public GridFrame(GridView gridView, JButton nextButton, JButton clearButton) {
 
-    public GridFrame(
-            Grid grid,
-            GridView gridView,
-            JButton nextButton
-    ) {
-        setSize(650, 350);
+        //encapsulate?
+        this.grid = gridView.grid;
+        this.nextButton = nextButton;
+        this.clearButton = clearButton;
+
+        setSize(1000, 600);
         setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
         setTitle("Game of Life");
         setLayout(new BorderLayout());
 
         add(gridView, BorderLayout.CENTER);
+        bottom.setLayout(new FlowLayout());
+        clearButton.setText("Clear");
+        bottom.add(clearButton);
+        nextButton.setText("Next");
         nextButton.addActionListener(ActionEvent -> {getNextGen(grid);});
+        bottom.add(nextButton);
+        add(bottom, BorderLayout.SOUTH);
+        nextButton.addActionListener(ActionEvent -> {
+            getNextGen(grid);
+        });
```

Comment:
Instead of calling a method in the same Frame class, you could just call in the actionListener ActionEvent grid.goToNextGeneration(); it will clean up the code and do the same thing.

Reply:
Sounds legit, but I don't want to alter the listener. @shweinfeld perhaps you can implement this?

Reply:
Yup.